### PR TITLE
Kubernetes 1.24: contextual logging blog post

### PR DESCRIPTION
**Pull request by pohly**
Contextual logging is an enhancement that landed as alpha in Kubernetes
1.24. Because it targets primarily developers of Kubernetes at this point,
publishing it under https://k8s.dev/blog on the same day as it also gets
published on the main blog (see
https://github.com/kubernetes/website/pull/32656) makes sense.

### CODE REVIEW

**Comment by pohly** [comment]
/hold

Because of https://github.com/kubernetes/contributor-site/issues/281

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (jberkus) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:14]: suggestion: put a call for "and you can join us and help" here too, not just at the bottom

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (jberkus) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:39]: so ... what happens if the message text contains a ">" ?

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (jberkus) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:251]: should have a link here to https://github.com/kubernetes/community/tree/master/wg-structured-logging

**Review by jberkus — APPROVED** [review]
A few minor suggestion, no blockers.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:39]: It will be indented, in contrast to the `>` end marker which isn't.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:14]: Makes sense, will change it.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:251]: Agreed.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:39]: I thought they were indented differently - let me double check.

Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:39]: Right, the value uses \t and the end marker a single space.

This isn't visible when looking at the diff, but renders okay when looking at the full file.

Comment (jberkus) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:39]: ah, I see, so the delimiters are really "<\n" and "^>".

Probably does *not* need to be added to the blog post, then.

Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:39]: Or to be very precise, `=<\n` and `^ >` (one space) with `^\t` for the lines in the middle :sweat_smile: 

Agreed, this doesn't need to be in the blog post.

**Comment by pohly** [comment]
Comments addressed, please take another look.

**Comment by pohly** [comment]
I have force-pushed [an update](https://github.com/kubernetes/contributor-site/compare/409eabb0b44ebc83a50f1978c6ec939073c79c09..5607a77ea042eb32ce16731a7608bf8915292f94) to address the review feedback from https://github.com/kubernetes/website/pull/32656#pullrequestreview-958739741

**Comment by jberkus** [comment]
/lgtm 

from me

**Comment by pohly** [comment]
The markup was wrong in at least one place and line breaks in link text looked odd, at least on GitHub. I've reformatted those.

When this lands in the blog, paragraphs will get reflown, right? I thought GitHub also did that, but apparently not.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:16]: (nit)
```suggestion
## Structured logging
```
(based on style guide for https://k8s.io/ and its blog)

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:62]: (nit)
```suggestion
## Contextual logging
```
(based on style guide for https://k8s.io/ and its blog)

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:197]:
```suggestion
### Reusable output test
```

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:236]:
```suggestion
We are in the process of [moving the tool]((https://github.com/kubernetes/klog/issues/312)
into a new Git repository, because it isn't
```

Previous link text suggested it would be a link to the new repo, but actually wasn't.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:227]:
```suggestion
 [`logcheck`](https://github.com/kubernetes/klog/tree/788efcdee1e9be0bfbe5b076343d447314f2377e/hack/tools/logcheck)
```
unless we are confident that the layout of `/hack` in that repo is fixed enough that this link won't go stale.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [inline content/en/blog/2022/2022-05-25-contextual-logging.md:137]: You could also write, eg:
```suggestion
The sample code for klog includes an an
[example](https://github.com/kubernetes/klog/blob/v2.60.1/ktesting/example/example_test.go)
for a unit test with per-test output.
```

which [uses the active voice](https://kubernetes.io/docs/contribute/style/style-guide/#use-active-voice).

**Review by sftim — COMMENTED** [review]
Here's a few, _small_ / not-merge-blocking, bits of feedback.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:236]: Accepted with minor syntax fix.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (pohly) [reply content/en/blog/2022/2022-05-25-contextual-logging.md:137]: Accepted with fix for "an an".

**Comment by pohly** [comment]
@sftim, @jberkus : we are getting closer to the intended publishing data (25th of May). Is this good to go as it is?

Can we merge it now or unhold it on the day when it is supposed to go live (https://github.com/kubernetes/contributor-site/issues/281)?

**Comment by sftim** [comment]
We wouldn't merge it now. We should get the approvals and reviews in place so that an unhold on the day of publication is all ready to go.

File: content/en/blog/2022/2022-05-25-contextual-logging.md
Comment (sftim) [reply content/en/blog/2022/2022-05-25-contextual-logging.md]: Ah yep. Thanks.

**Comment by sftim** [comment]
Is https://github.com/kubernetes/website/pull/32656 level with the article text as reviewed here?

**Comment by pohly** [comment]
> Is [kubernetes/website#32656](https://github.com/kubernetes/website/pull/32656) level with the article text as reviewed here?

Not yet, but as this seems to be the final version now I'll copy it over.

**Comment by pohly** [comment]
/hold cancel

Let's publish...

**Pull request merged**
